ui/ui-components/pages/model/[modelId]/overview.tsx

```diff
@@ -0,0 +1,168 @@
+import ModelOverviewSecondarySources from "../../../components/model/overview/ModelOverviewSecondarySources";
+import PageHeader from "../../../components/PageHeader";
+import { GrouparooModelContextProvider } from "../../../contexts/grouparooModelContext";
+import { UseApi } from "../../../hooks/useApi";
+import { Actions, Models } from "../../../utils/apiData";
+
+interface ModelOverviewCardProps {
+  title: string;
+}
+
+const ModelOverviewCard: React.FC<ModelOverviewCardProps> = ({
+  title,
+  children,
+}) => {
+  return (
+    <Card>
+      <Card.Header as="h5">{title}</Card.Header>
+      {children}
+    </Card>
+  );
+};
+
+interface Props {
+  model: Models.GrouparooModelType;
+  primarySource?: Models.SourceType;
+  secondarySources: Models.SourceType[];
+  groups: Models.GroupType[];
+  schedules: Models.ScheduleType[];
+  destinations?: Models.DestinationType[];
+}
+
+const Page: NextPage<Props & { ctx: any; errorHandler: any }> = ({
+  ctx,
+  errorHandler,
+  model,
+  primarySource,
+  secondarySources,
+  groups,
+  schedules,
+  destinations,
+}) => {
+  const { execApi } = UseApi(ctx, errorHandler);
+  const sources = useMemo(
+    () => [primarySource, ...secondarySources],
+    [primarySource, secondarySources]
+  );
+
+  return (
+    <GrouparooModelContextProvider model={model}>
+      <Head>
+        <title>Grouparoo: {model.name}</title>
+      </Head>
+
+      <Container>
+        <PageHeader
+          iconType="grouparooModel"
+          icon={model.icon}
+          title={model.name}
+        />
+        <p>
+          Define your {model.name} model here with sources, groups, and
+          destinations.
+        </p>
+
+        <Row className="mb-4">
+          <Col>
+            <ModelOverviewCard title="Model Data">
+              <ListGroup className="list-group-flush">
+                <ListGroupItem>
+                  <ModelOverviewPrimarySource source={primarySource} />
+                </ListGroupItem>
+
+                <ListGroupItem>
+                  <ModelOverviewSecondarySources sources={secondarySources} />
+                </ListGroupItem>
+
+                <ListGroupItem>
+                  <ModelOverviewGroups groups={groups} />
+                </ListGroupItem>
+
+                <ListGroupItem>
+                  <ModelOverviewSchedules
+                    schedules={schedules}
+                    sources={sources}
+                    execApi={execApi}
+                  />
+                </ListGroupItem>
+              </ListGroup>
+            </ModelOverviewCard>
+          </Col>
+        </Row>
+        <Row>
+          <Col>
+            <ModelOverviewCard title="Destinations">
+              <ModelOverviewDestinations destinations={destinations} />
+            </ModelOverviewCard>
+          </Col>
+        </Row>
+      </Container>
+    </GrouparooModelContextProvider>
+  );
+};
+
+export const getServerSideProps: GetServerSideProps<Props> = async (
```

Comment:
This uses `getServerSideProps` over `getInitialProps` to move away from deprecated features. However, you can see that in ui-enterprise the import gets a little more complex

https://nextjs.org/docs/api-reference/data-fetching/getInitialProps